tlvc/arch/i86/drivers/block/directhd.c

```diff
@@ -431,47 +467,71 @@ int INITPROC directhd_init(void)
 	 * of them even have more than 16.384 cylinders
 	 *
 	 * This is some sort of bugfix, we will use same method as real Linux -
-	 * work with disk geometry set in current translation mode rather than
-	 * using physical drive info. Physical info might correspond to logical
-	 * info for some drives (those which don't support/use LBA mode).
+	 * work with disk geometry set in current translation mode (54-56) if valid (53)
+	 * rather than physical drive info. Old drives have only physical, which may
+	 * be misleading: The copnmfigured BIOS drive type's CHS mapping may be different.
+	 * E.g the Compaq drive type 17 (conner 42MB) reports 806/4/26 while the BIOS
```

Comment:
Are you saying here that the "BIOS configured value" (assume from user or computer manufacturer configuration) should always override what the drive itself might report (if indeed it reports anything)? If both work, what is the reason that drive and BIOS CHS may be different? 

It would seem then for TLVC that a CHS specification (likely from /bootopts?) would just have to match the way that TLVC calculates CHS from block numbers. (?)

In the past, I have only noticed possibly a cylinder number being 1 greater through the IDE probing than the BIOS-reported value.